src/Services/QueuedJobService.php

```diff
@@ -235,64 +235,30 @@ public function queueJob(QueuedJob $job, $startAfter = null, $userId = null, $qu
     {
         $signature = $job->getSignature();
 
-        // see if we already have this job in a queue
-        $filter = [
-            'Signature' => $signature,
-            'JobStatus' => [
-                QueuedJob::STATUS_NEW,
-                QueuedJob::STATUS_INIT,
-            ],
-        ];
-
-        $existing = QueuedJobDescriptor::get()
-            ->filter($filter)
-            ->first();
-
-        if ($existing && $existing->ID) {
-            return $existing->ID;
-        }
-
-        $jobDescriptor = new QueuedJobDescriptor();
-        $jobDescriptor->JobTitle = $job->getTitle();
-        $jobDescriptor->JobType = $queueName ? $queueName : $job->getJobType();
-        $jobDescriptor->Signature = $signature;
-        $jobDescriptor->Implementation = get_class($job);
-        $jobDescriptor->StartAfter = $startAfter;
-
-        // no user provided - fallback to job user default
-        if ($userId === null && $job instanceof UserContextInterface) {
-            $userId = $job->getRunAsMemberID();
-        }
+        // Create the initial object
+        $jobDescriptor = $this->createJobDescriptor($job, $signature, $startAfter, $userId, $queueName);
 
-        // still no user - fallback to current user
-        if ($userId === null) {
-            if (Security::getCurrentUser() && Security::getCurrentUser()->exists()) {
-                // current user available
-                $runAsID = Security::getCurrentUser()->ID;
-            } else {
-                // current user unavailable
-                $runAsID = 0;
-            }
-        } else {
-            $runAsID = $userId;
+        try {
+            return $this->findOrMakeJobDescriptorFromSignature($signature, $job, $jobDescriptor, $startAfter);
+        } catch (\Throwable $e) {
+            // note that error here may not be an issue as failing to acquire a job lock is a valid state
+            // which happens when other process claimed the job lock first
```

Comment:
I think the comments should be updated to describe the job signature, not the job lock as that is a different feature. Please also update the log message so this feature is not confused with the job locking.